### PR TITLE
Adding the Cheng-Todreas correlation for the friction factor

### PR DESCRIPTION
**Pull request by luizaldeia**
Refs #24340

### CODE REVIEW

File: modules/thermal_hydraulics/test/tests/materials/wall_friction_cheng/tests [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+  design = 'ADWallFrictionChengMaterial.md'
+  issues = '#24340'
+  [normal_operation_square_array_laminar]
+    requirement = "The system shall compute the friction factor when the Cheng-Todreas correlation "
+                  "is used with:"
+    [PoD_105_interior_sqr_lam]
+      type = 'CSVDiff'
+      input = 'cheng_ff_test.i'
+      csvdiff = 'sqr_lam_105_int.csv'
+      detail = 'an interior subchannel, with a square array, laminar flow, and a Pitch-to-Diameter ratio below 1.10 and above 1.0,'
```
Comment (licharlot) [inline modules/thermal_hydraulics/test/tests/materials/wall_friction_cheng/tests:11]: remove above 1, here and below

File: modules/thermal_hydraulics/include/materials/ADWallFrictionChengMaterial.h [hunk 2]
```diff
@@ -0,0 +1,63 @@
+//* This file is part of the MOOSE framework
```
Comment (licharlot) [inline modules/thermal_hydraulics/include/materials/ADWallFrictionChengMaterial.h:1]: Implement similar changes as suggested by Josh on your HTC PR.

**Comment by loganharbour** [comment]
Welcome to the occasional civet infinite loop puke

**Comment by luizaldeia** [comment]
Ahahahahaha thanks, I ended up doing my fixup on the wrong commit and ended up here

**Pull request closed without merging**